includes/Modules/Analytics_4/Report.php

```diff
@@ -47,131 +41,8 @@ public function __construct( Context $context ) {
-	 */
-	protected function parse_dateranges( Data_Request $data ) {
-		$date_ranges = array();
-		$start_date  = $data['startDate'];
-		$end_date    = $data['endDate'];
-		if ( strtotime( $start_date ) && strtotime( $end_date ) ) {
-			$compare_start_date = $data['compareStartDate'];
-			$compare_end_date   = $data['compareEndDate'];
-			$date_ranges[]      = array( $start_date, $end_date );
-
-			// When using multiple date ranges, it changes the structure of the response:
-			// Aggregate properties (minimum, maximum, totals) will have an entry per date range.
-			// The rows property will have additional row entries for each date range.
-			if ( strtotime( $compare_start_date ) && strtotime( $compare_end_date ) ) {
-				$date_ranges[] = array( $compare_start_date, $compare_end_date );
-			}
-		} else {
-			// Default the date range to the last 28 days.
-			$date_ranges[] = Date::parse_date_range( 'last-28-days', 1 );
-		}
-
-		$date_ranges = array_map(
-			function ( $date_range ) {
-				list ( $start_date, $end_date ) = $date_range;
-				$date_range                     = new Google_Service_AnalyticsData_DateRange();
-				$date_range->setStartDate( $start_date );
-				$date_range->setEndDate( $end_date );
-
-				return $date_range;
-			},
-			$date_ranges
-		);
-
-		return $date_ranges;
-	}
-
-	/**
-	 * Parses the orderby value of the data request into an array of AnalyticsData OrderBy object instances.
-	 *
-	 * @since 1.99.0
-	 *
-	 * @param Data_Request $data Data request object.
-	 * @return Google_Service_AnalyticsData_OrderBy[] An array of AnalyticsData OrderBy objects.
-	 */
-	protected function parse_orderby( Data_Request $data ) {
-		$orderby = $data['orderby'];
-		if ( empty( $orderby ) || ! is_array( $orderby ) || ! wp_is_numeric_array( $orderby ) ) {
-			return array();
-		}
-
-		$results = array_map(
-			function ( $order_def ) {
-				$order_by = new Google_Service_AnalyticsData_OrderBy();
-				$order_by->setDesc( ! empty( $order_def['desc'] ) );
-
-				if ( isset( $order_def['metric'] ) && isset( $order_def['metric']['metricName'] ) ) {
-					$metric_order_by = new Google_Service_AnalyticsData_MetricOrderBy();
-					$metric_order_by->setMetricName( $order_def['metric']['metricName'] );
-					$order_by->setMetric( $metric_order_by );
-				} elseif ( isset( $order_def['dimension'] ) && isset( $order_def['dimension']['dimensionName'] ) ) {
-					$dimension_order_by = new Google_Service_AnalyticsData_DimensionOrderBy();
-					$dimension_order_by->setDimensionName( $order_def['dimension']['dimensionName'] );
-					$order_by->setDimension( $dimension_order_by );
-				} else {
-					return null;
-				}
-
-				return $order_by;
-			},
-			$orderby
-		);
-
-		$results = array_filter( $results );
-		$results = array_values( $results );
-
-		return $results;
-	}
+	// NOTE: The majority of this classes logic has been abstracted to
+	// ReportParsers which contains the shared methods for both
+	// Report and PivotReport classes.
```

Comment:
What is even really left of this class? It seems like it isn't implementing anything itself… why keep it around? 🤔

Reply:
IMO we should keep this class around for a consistant structure between both report types and in theory we can use this class in future to implement any non pivot report specific logic.

Reply:
Alrighty, works for me 👍🏻